app/src/main/java/com/gravatar/app/AppTrackerSetupDataProvider.kt

```diff
@@ -0,0 +1,31 @@
+package com.gravatar.app
+
+import com.gravatar.analytics.TrackerSetupData
+import com.gravatar.analytics.TrackerSetupDataProvider
+import com.gravatar.analytics.TrackingState
+import com.gravatar.app.usercomponent.domain.repository.UserRepository
+import com.gravatar.app.usercomponent.domain.usecase.GetPrivacySettings
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+
+class AppTrackerSetupDataProvider(
+    private val getPrivacySettings: GetPrivacySettings,
+    private val userRepository: UserRepository,
+) : TrackerSetupDataProvider {
+    override fun getTrackerSetupData(): Flow<TrackerSetupData> {
+        val userIdFlow: Flow<String?> = userRepository.getProfile().map { it?.userId?.toString() }.distinctUntilChanged()
```

Comment:
🛑 We need to use the WPCOM userName for tracks instead of the userId.

```suggestion
        val userIdFlow: Flow<String?> = userRepository.getProfile().map { it?.userLogin }.distinctUntilChanged()
```

That suggestion is not enough, as we need to persist it in the DB.